assembly/codegen/cli/ext/tpl/main.tpl

```diff
@@ -1,43 +1,37 @@
-function isSelectorEqual(l: u8[], r: u8[]): boolean {
-  if (l.length != r.length) return false;
-  for (let i = 0; i < l.length; i++) {
-    if (l[i] != r[i]) return false;
-  }
-  return true;
-}
+var msg: Msg = new Msg();
 
 export function deploy(): i32 {
-  // const selector = arryToHexString(fnSelector);
-  // Log.println("deploy.fnSelctor: " + selector);
-
-  const ctorWithParams: u8[] = {{ctorWithParamsVal}}; // 
-  const ctorWithoutParams: u8[] = {{ctorWithoutParamsVal}}
-
-  let _{{exportDef.className}} = new {{exportDef.className}}();
+  let _{{exportDef.contractName}} = new {{exportDef.className}}();
 
   {{#each exportDef.deployers}}
-  if (msg.isSelector(ctorWithParams)) {
-    _{{../exportDef.className}}.{{methodName}}(false);
+  const {{methodName}}Selector: u8[] = {{#selectorArr methodName}}{{/selectorArr}};
+  if (msg.isSelector({{methodName}}Selector)) {
+    const fnParameters = new FnParameters(msg.data);
+    {{#each paramters}}
+    let p{{_index}} = fnParameters.get<{{codecType}}>();
+    {{/each}}
+    _{{../exportDef.contractName}}.{{methodName}}({{#joinParams paramters}}{{/joinParams}}{{ctrDefaultVals}});
```

Comment:
We should check if there are parameters defined in a function, and then we can do instantiating operations.

For saving gas.